### ISSUE TITLE
Too long filenames for Windows 10

### ISSUE THREAD

**Issue by ErikBrown2** [sabnzbd/sabnzbd#1746]
Recently I noticed that I get often too long filenames in completed folders. Windows explorer does not allow me to rename them.

The default for the maximum folder name length in Settings-Special-max_foldername_length is 246. This appears to be much too long. 

As a test to see how long folder names can be, I opened notepad and typed 242 characters, then created a folder in the root of c: and pasted the 242 characters in as folder name. Then I checked the name of the created folder in properties and the name has a length of 226 characters. If I try to create a sub folder in this folder then I directly get a message that the filename (?) would be too long. When I copy a file into this folder, then this works. But when I try to rename the file, then this is rejected. The message is that the filename is too long. So I think that the default of 246 for the maximum folder length should be changed to anyway not more than 226. But I have now set it on my PC to 128 to be on the safe side.

I use Windows 10 pro 20H2, SABnzbd 3.2.0Beta1 and I have enabled long filenames in Windows

**Comment by Safihre** [comment]
Currently there's no limit on filenames, something we also noticed and @sanderjo was trying to patch in #1721. The setting you edited is for the folder name.

**Comment by Safihre** [comment]
Strange, I expected enabling long filenames in Windows would not give these problems. Are you sure you have it enabled?

In any case, this was how it was designed to be. Shortening the filenames has also given us plenty of weirdness over the past years.

**Comment by thezoggy** [comment]
also its a mixture of windows + python + filesystem.  let alone sab coding and use of win32api, then also its utilities. for awhile the limiting factor was unrar as it didnt support long file notation wile the rest of sab did, so we had to artificially limit this for its sake. then there is the whole running sab on one file system but then the complete folder being on different one.. or encrypted systems which all can play a factor.  some additional info: https://github.com/sabnzbd/sabnzbd/issues/1597

TL;DR; if you are having issues you need to share logs so we can see why your setup is causing problems and make suggestions on how to fix. also you can mask most path issues by just not using a download path thats not several folders deep (since each part can only be so much and the total path can only be so much).
